stream: add AbortSignal support to promisified pipeline

Linkgoron opened the pull request:
This PR adds support for AbortSignal to promisified pipeline (in `stream/promises`)

This resolves #37321. 

As an aside, this wouldn't be hard to actually push it into pipeline itself and support it in the regular version, however it's already pretty simple to just call `addAbortSignal` from `stream` on the received pipeline.

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x]  commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

benjamingr commented on test/parallel/test-stream-pipeline.js:
```diff
@@ -469,6 +469,77 @@ const net = require('net');
+  // Check pre-aborted signal
+  const pipelinePromise = promisify(pipeline);
+  async function run() {
+    const signal = new EventTarget();
```
Any reason this test isn't with an AbortController?

Linkgoron replied:
not really

benjamingr commented:
@ronag

ronag commented:
@benjamingr: Isn't the usual pattern to send an options object i.e. `{ signal }`?

ronag reviewed (changes requested):
needs docs?

benjamingr commented:
Yeah I tend to prefer options with a signal too

Linkgoron commented:
I've updated the docs, and also changed it to receive an `options` parameter, in the promise version.

@ronag I could also "push" the changes from the promise version, into the regular pipeline implementation, so that both the "regular" pipeline and promisified pipeline support cancellation. where the API will change to:
`stream.pipeline(source[, ...transforms], destination[,options], callback)`
`stream.pipeline(streams[,options], callback)`
 
Similarly to what was done in `readline.question`, [a few weeks ago](https://github.com/nodejs/node/commit/19981ed76eee180ac75e08409d7f3e9db673c841).

benjamingr commented:
Landed in 38f6e5a789b1 🎉
